File: Homeworks and Labs/Arrays-Exercises/11.EqualSums/EqualSums.cs
using System;
using System.Linq;


class EqualSums
{
    static void Main(string[] args)
    {
        int[] numbers = Console.ReadLine().Split().Select(int.Parse).ToArray();

        bool hasEqualSum = false;

        for (int i = 0; i < numbers.Length; i++)
        {
            int leftSum = numbers.Take(i).Sum();
            int rightSum = numbers.Skip(i + 1).Sum();
            if (leftSum == rightSum)
            {
                Console.WriteLine(i);
                hasEqualSum = true;
                break;
            }
        }

        if (!hasEqualSum)
        {
            Console.WriteLine("no");
        }
    }
}

File: Software Technologies/Java Basics - Exercises/src/EqualSums.java
import java.util.Scanner;

public class EqualSums {
    public static void main(String [] args)
    {
        Scanner scanner = new Scanner(System.in);
        String [] numbersAsStr = scanner.nextLine().split("\\s");

        int[] numbers = new int[numbersAsStr.length];

        for (int i = 0; i < numbersAsStr.length; i++) {
            numbers[i]=Integer.parseInt(numbersAsStr[i]);
        }

        boolean hasEqualSum = false;

        for (int i = 0; i < numbers.length; i++) {
            int leftSum = 0;
            for (int j = 0; j < i; j++) {
                leftSum+=numbers[j];
            }

            int rightSum = 0;
            for (int j = i+1; j < numbers.length; j++) {
                rightSum+=numbers[j];
            }

            if (leftSum == rightSum) {
                System.out.println(i);
                hasEqualSum = true;
                break;
            }
        }

        if (!hasEqualSum) {
            System.out.println("no");
        }
    }
}
